models/sharky/solana/sharky_solana_events.sql

```diff
@@ -35,107 +30,333 @@ WITH sharky_txs AS (
             blockchain IS NULL
             AND symbol = 'SOL'
             {% if not is_incremental() %}
-            AND minute >= '{{ project_start_date }}'
+            AND minute >= TIMESTAMP '{{ project_start_date }}'
             {% else %}
-            AND minute >= date_trunc("day", now() - interval '1 week')
+            AND minute >= DATE_TRUNC('day', NOW() - INTERVAL '7' DAY)
             {% endif %}
     ),
-    filtered_txs AS (
-        SELECT signatures,
-               block_date,
-               block_time,
-               block_slot,
-               post_balances,
-               pre_balances,
-               instructions,
-               signer,
-               id
-        FROM {{ source('solana','transactions') }} tx
+    offers AS (
+        SELECT
+            call_block_date,
+            call_block_slot,
+            call_block_time,
+            call_tx_signer,
+            call_tx_id,
+            account_loan,
+            'Offer' AS evt_type,
+            account_escrow,
+            account_orderBook,
+            NULL AS account_collateralMint,
+            ARRAY[CAST(ROW(call_data, call_account_arguments) AS ROW(data VARBINARY, account_arguments ARRAY<VARCHAR>))] AS sharky_instructions,
+            CAST(principalLamports AS BIGINT) AS amount_raw
```

Comment:
suggestion: use dunesql datatype `uint256` instead? or is that just an evm concept and solana would prefer bigint?

this would apply to raw amounts throughout, just an open question

Reply:
if it's a smaller data type we can consider using `uint256` here

Reply:
there is also `int256`
https://dune.com/docs/query/DuneSQL-reference/datatypes/#int256-dune-sql

again -- just a suggestion, as this is a standalone spell. if in the future it rolls into a sector-level spell, we would look to standardize

Reply:
I would keep it like this for now and if a best practice arises I will take care of updating it to follow them 👍🏾

Reply:
I would keep it like this for now and if a best practice arises I will take care of updating it to follow them 👍🏾

Reply:
sounds good!